gusto/physics.py

```diff
@@ -1751,3 +1751,72 @@ def evaluate(self, x_in, dt):
 
         self.X.assign(x_in)
         self.rho_recoverer.project()
+
+
+class TerminatorToy(PhysicsParametrisation):
+    """
+    Setup the Terminator Toy chemistry interaction
+    as specified in 'The terminator toy chemistry test ...'
+    Lauritzen et. al. (2014).
+
+    The coupled equations for the two species are given by:
+
+    D/Dt (X) = 2Kx
+    D/Dt (X2) = -Kx
+
+    where Kx = k1*X2 - k2*(X**2)
+    """
+
+    def __init__(self, equation, k1=1, k2=1,
+                 species1_name='X', species2_name='X2'):
+        """
+        Args:
+            equation (:class: 'PrognosticEquationSet'): the model's equation
+            k1(float, optional): Reaction rate for species 1 (X). Defaults to a constant
+              over the domain.
+            k2(float, optional): Reaction rate for species 2 (X2). Defaults to a constant
+              over the domain.
+            species1_name(str, optional): Name of the first interacting species. Defaults
+              to 'X'.
+            species2_name(str, optional): Name of the second interacting species. Defaults
+              to 'X2'.
+        """
+
+        label_name = 'terminator_toy'
+        super().__init__(equation, label_name, parameters=None)
+
+        assert species1_name in equation.field_names, f"Field {species1_name} does not exist in the equation set"
```

Comment:
Very picky point here. It might be slightly better if we change this error (and the one below) to something like:
```
if species1_name not in equation.field_names:
    raise ValueError(...)
```
as this is the normal type of python error when the wrong argument is given to a routine

Reply:
Cool, I've done that for the two name checks. Is the check for the function spaces ok as it is, as this is not an argument error, but a compatibility error?

Reply:
I think the check is OK as is, as you say it is not about having an incorrect argument but that the model set up isn't correct